lib/src/standalone.dart

```diff
@@ -0,0 +1,203 @@
+
+  if (File("LICENSE").existsSync()) {
+    archive.addFile(file("$standaloneName/src/LICENSE", "LICENSE"));
+  }
+
+  for (var entrypoint in entrypoints) {
+    var basename = p.basename(entrypoint);
+    archive.addFile(file(
+        "$standaloneName/src/$basename.snapshot",
+        _useNative(os, x64: x64)
+            ? "build/$basename.native"
+            : "build/$basename.snapshot"));
+  }
+
+  // Do this separately from adding entrypoints because multiple executables may
+  // have the same entrypoint.
+  executables.forEach((name, path) {
+    archive.addFile(fileFromString(
+        "$standaloneName/$name${os == 'windows' ? '.bat' : ''}",
+        renderTemplate(
+            "standalone/executable.${os == 'windows' ? 'bat' : 'sh'}", {
+          "name": standaloneName,
+          "version": _useNative(os, x64: x64) ? null : version.toString(),
+          "executable": p.basename(path)
+        }),
+        executable: true));
+  });
+
+  var prefix = 'build/$standaloneName-$version-$os-${_arch(x64)}';
+  if (os == 'windows') {
+    var output = "$prefix.zip";
+    log("Creating $output...");
+    File(output).writeAsBytesSync(ZipEncoder().encode(archive));
+  } else {
+    var output = "$prefix.tar.gz";
+    log("Creating $output...");
+    File(output)
+        .writeAsBytesSync(GZipEncoder().encode(TarEncoder().encode(archive)));
+  }
+}
+
+/// Returns the binary contents of the `dart` or `dartaotruntime` exectuable for
+/// the given [os] and architecture.
+Future<List<int>> _dartExecutable(String os, {@required bool x64}) async {
+  // If we're building for the same SDK we're using, load its executable from
+  // disk rather than downloading it fresh.
+  if (_useNative(os, x64: x64)) {
+    return File(p.join(
+            sdkDir.path, "bin/dartaotruntime${os == 'windows' ? '.exe' : ''}"))
+        .readAsBytesSync();
+  } else if (isTesting) {
+    // Don't actually download full SDKs in test mode, just return a dummy
+    // executable.
+    return utf8.encode("Dart $os ${_arch(x64)}");
+  }
+
+  // TODO(nweiz): Compile a single executable that embeds the Dart VM and the
+  // snapshot when dart-lang/sdk#27596 is fixed.
+  var channel = isDevSdk ? "dev" : "stable";
+  var url = "https://storage.googleapis.com/dart-archive/channels/$channel/"
+      "release/$dartVersion/sdk/dartsdk-$os-${_arch(x64)}-release.zip";
+  log("Downloading $url...");
+  var response = await client.get(Uri.parse(url));
+  if (response.statusCode ~/ 100 != 2) {
+    fail("Failed to download package: ${response.statusCode} "
+        "${response.reasonPhrase}.");
+  }
+
+  var filename = "/bin/dart${os == 'windows' ? '.exe' : ''}";
+  return ZipDecoder()
+      .decodeBytes(response.bodyBytes)
+      .firstWhere((file) => file.name.endsWith(filename))
+      .content as List<int>;
+}
+
+/// Returns the architecture name for the given boolean.
+String _arch(bool x64) => x64 ? "x64" : "ia32";
```

Comment:
binary extension might be another good helper here. `os == 'windows' ? '.exe' : ''` appears three times in this file, and all three are in string interpolations. :/

Reply:
Done.